onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -109,10 +109,10 @@ std::string nuphar_settings;
 #endif
 
 namespace onnxruntime {
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CPU(int use_arena);
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CUDA(int device_id);
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Tensorrt(int device_id);
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Dnnl(int use_arena);
+std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CPU(bool use_arena);
+std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CUDA(int device_id, bool use_cuda_arena, bool use_cpu_arena);
```

Comment:
nit: easy to mess up (reverse) the values of these (use_cuda_arena and use_cpu_arena) at the call site.

Reply:
Do you want them combined into an enum? At least it's slightly clearer than the BrainSlice one. :)

---
In reply to: [372065123](https://github.com/microsoft/onnxruntime/pull/2850#discussion_r372065123) [](ancestors = 372065123)

Reply:
Up to you. Doesn't have to be this PR. The BrainSlice interface is poorly designed and is not a good reference. Just for reference: https://www.aristeia.com/Papers/IEEE_Software_JulAug_2004_revised.htm